core/src/main/scala/org/apache/spark/network/Security.scala

```diff
@@ -35,33 +37,15 @@ object Security extends Logging {
     }
   }
 
-  def enableSSL(spark: SparkSession, conf: H2OConf): Unit = enableSSL(spark, conf.sparkConf)
+  def enableSSL(spark: SparkSession, conf: H2OConf): Unit = enableSSL(conf)
 
-  def enableSSL(spark: SparkSession): Unit = enableSSL(spark, spark.sparkContext.conf)
-
-  def enableFlowSSL(spark: SparkSession, conf: H2OConf): H2OConf = {
+  def enableFlowSSL(conf: H2OConf): H2OConf = {
     val sslPair = SparklingWaterSecurityUtils.generateSSLPair(namePrefix = "h2o-internal-auto-flow-ssl")
     conf.setJks(sslPair.jks.getLocation)
     conf.setJksPass(sslPair.jks.pass)
   }
 
-  def enableSSL(sc: SparkContext, conf: SparkConf): Unit = {
-    logWarning("Method Security.enableSSL with an argument of type SparkContext is deprecated and " +
-      "parameter of type SparkSession is preferred.")
-      enableSSL(SparkSession.builder().sparkContext(sc).getOrCreate(), conf)
-  }
-
-  def enableSSL(sc: SparkContext): Unit = {
-    logWarning("Method Security.enableSSL with an argument of type SparkContext is deprecated and " +
-      "parameter of type SparkSession is preferred.")
-    enableSSL(SparkSession.builder().sparkContext(sc).getOrCreate(), sc.conf)
-  }
-
-  def enableSSL(sc: SparkContext, conf: H2OConf): Unit = {
-    logWarning("Method Security.enableSSL with an argument of type SparkContext is deprecated and " +
-      "parameter of type SparkSession is preferred.")
-    enableSSL(SparkSession.builder().sparkContext(sc).getOrCreate(), conf.sparkConf)
-  }
-
-
+  def enableFlowSSL(spark: SparkSession, conf: H2OConf): H2OConf = enableFlowSSL(conf)
```

Comment:
Shouldn't we deprecate this method overloading?

Reply:
DTTO ^